tests/integration/test_transformers.py

```diff
@@ -0,0 +1,92 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+import rdt.transformers
+import tests.datasets
+
+DATA_SIZES = [100, 1000, 10000]
+
+# Temporary mapping
+TRANSFORMER_TO_TYPE = {
+    rdt.transformers.null.NullTransformer: None,
+    rdt.transformers.boolean.BooleanTransformer: 'boolean',
+    rdt.transformers.categorical.CategoricalTransformer: 'categorical',
+    rdt.transformers.categorical.OneHotEncodingTransformer: 'categorical',
+    rdt.transformers.categorical.LabelEncodingTransformer: 'categorical',
+    rdt.transformers.datetime.DatetimeTransformer: 'datetime',
+    rdt.transformers.numerical.NumericalTransformer: 'numerical',
+    rdt.transformers.numerical.GaussianCopulaTransformer: 'numerical',
+}
+
+
+def get_subclasses(obj):
+    subclasses = obj.__subclasses__()
+
+    if len(subclasses) == 0:
+        return []
+
+    for sc in subclasses:
+        subclasses.extend(get_subclasses(sc))
+
+    return subclasses
+
+
+def build_generator_map():
+    generators = {}
+
+    for g in tests.datasets.BaseDatasetGenerator.__subclasses__():
+        generators[g.DATA_TYPE] = g.__subclasses__()
+
+    return generators
+
+
+def get_transformer_test_cases():
+    test_cases = []
+
+    transformers = get_subclasses(rdt.transformers.BaseTransformer)
+    generators = build_generator_map()
+
+    for t in transformers:
+        for ds in DATA_SIZES:
+            dt = TRANSFORMER_TO_TYPE[t]
+
+            if dt is None:
+                continue
+
+            gs = generators[dt]
+
+            for g in gs:
+                test_cases.append((t, ds, g))
+
+    return test_cases
+
+
+test_cases = get_transformer_test_cases()
+
+
+@pytest.mark.parametrize('transformer,data_size,dataset_generator', test_cases)
+def test_transformer(transformer, data_size, dataset_generator):
```

Comment:
I added a comment about this directly in the issue, so we can discuss there in a more organized way.